src/security-archive/src/index.ts

```diff
@@ -0,0 +1,297 @@
+// eslint-disable-next-line import/no-unresolved
+import { DatabaseSync } from 'node:sqlite'
+import { LRUCache } from 'lru-cache'
+import { loadPackageJson } from 'package-json-from-dist'
+import { asDepID, splitDepID, joinDepIDTuple } from '@vltpkg/dep-id'
+import { error } from '@vltpkg/error-cause'
+import { XDG } from '@vltpkg/xdg'
+import { asPackageReportData } from './types.ts'
+import type { DepID } from '@vltpkg/dep-id'
+import type { GraphLike } from '@vltpkg/graph'
+import type { SpecOptions } from '@vltpkg/spec'
+import type {
+  PackageReportData,
+  SecurityArchiveLike,
+  SecurityArchiveRefreshOptions,
+} from './types.ts'
+
+const SOCKET_API_V0_URL = 'https://api.socket.dev/v0/purl?alerts=true'
+const SOCKET_PUBLIC_API_TOKEN =
+  'sktsec_t_--RAN5U4ivauy4w37-6aoKyYPDt5ZbaT5JBVMqiwKo_api'
+
+export const targetSecurityRegisty = 'https://registry.npmjs.org/'
+
+export type JSONItemResponse = {
+  namespace?: `@{string}`
+  name: string
+  version: string
+}
+
+export type LoadedEntry = {
+  ttl: LRUCache.Milliseconds
+  start: LRUCache.Milliseconds
+  value: PackageReportData
+}
+
+export type SecurityArchiveOptions = {
```

Comment:
This should extend `LRUOptions`, so that we can set the max, ttl, etc.

If anything really should not be set, then you can make it only settable to `undefined` in this type, like `fetchMethod?: undefined`.

Reply:
I'll go ahead and add that suggestion to the implementation to be explicit about not supporting `fetchMethod`.